src/Language/Haskell/Liquid/Constraint/Generate.hs

```diff
@@ -1028,6 +1022,39 @@ castTy g t (Tick _ e)
 castTy _ _ e
   = panic Nothing $ "castTy cannot handle expr " ++ showPpr e
 
+isClassConCo :: Coercion -> Maybe (Expr Var -> Expr Var)
+-- See Note [Type classes with a single method]
```

Comment:
@ranjitjhala @nikivazou GHC has this wonderful system of long-form comments in their source code that explain complicated checks and processes. It works as follows:

1. Write a self-contained comment describing at length the process, and give it a header

    ```
    -- Note [My complicated thing]
    -- details...
    ```

2. Reference `Note [My complicated thing]` at the relevant points of the implementation.

That's it! Now, someone else can come along, read the code, and grep for `Note [My complicated thing]` when they inevitably get confused.